Finish Creating Dashboard Screen

YANLING-H opened the pull request:
### Tasks completed in the PR
- [x] Finished implementing three types of interview card composable and dashboard screen.
- [x] Added two drawables and two pairs of strings for empty notes and empty resource screens.


### Screenshot of Android Studio Design
<img width="200" alt="Screen Shot 2023-03-29 at 12 26 41 AM" src="https://user-images.githubusercontent.com/67870437/228433642-334aa13d-93aa-495f-a553-34e72dc34142.png">

### Screenshot of the Emulator
<img width="200" alt="Screen Shot 2023-03-29 at 12 26 41 AM" src="https://user-images.githubusercontent.com/67870437/228433822-e08e3752-21f5-47c9-8065-ff2e3c1a11f8.png">

 The elevation seems awkward but I followed the [Material3.elevatedCard](https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#ElevatedCard(androidx.compose.ui.Modifier,androidx.compose.ui.graphics.Shape,androidx.compose.material3.CardColors,androidx.compose.material3.CardElevation,kotlin.Function1))

YANLING-H committed:
finish

YANLING-H committed:
format

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/data/model/Interview.kt:
```diff
@@ -2,11 +2,47 @@ package com.twain.interprep.data.model
-)
\ No newline at end of file
+)
+
```
I was trying to put define the data class `interview` here, but I didn't completely figure how to store `Calendar` object to the database. To focus on the components, I temporarily move the `Interview` data class to `InterviewCard.kt`

Audienix replied:
Check the class DateConverter in the code. I have updated the implementation and right way of storing Date object in Room DB.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/components/InterviewCard.kt:
```diff
@@ -0,0 +1,206 @@
+                    verticalArrangement = Arrangement.Center
+                ) {
+                    Text(
+                        text = SimpleDateFormat("dd").format(interview.date.time),
```
I didn't find a better way to format timestamps. I searched around and used `SimpleDateFormat` with `Calendar` object to format the time.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/theme/Color.kt:
```diff
@@ -5,4 +5,18 @@ import androidx.compose.ui.graphics.Color
+val DarkGray = Color(0xFF3F3D56)
+val Gray = Color(97, 97, 97, 64)
+val DateTimeGray = Color(0xFF625B71)
+val BrightGreen = Color(0xFF09AF00)
```
I understand that the name of those color doesn't make sense, but I don't know which naming format should I follow

Audienix replied:
I have updated. Plz follow the same.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/components/Header.kt:
```diff
@@ -0,0 +1,44 @@
+    modifier: Modifier,
+    fontWeight: FontWeight
+) {
+    Text(
```
Currently, I assume that the font in the whole app is the same and is the default one for simplicity.

Audienix replied:
Font is not the same in Figma, but for now let's not bother about special fonts. It's okay to go with default font.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt:
```diff
@@ -41,7 +59,74 @@ fun DashboardScreen(
+                                    horizontalArrangement = Arrangement.spacedBy(16.dp),
+                                    verticalAlignment = Alignment.CenterVertically
+                                ) {
+                                    items(5) { UpcomingInterviewCard() }
```
To focus on implementing each composable, I faked the InterviewCard data

Audienix replied:
I would suggest move all mock data in future PRs to a separate file, so that it would be easy to delete all mock data when we will have actual implementation.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/components/InterviewCard.kt:
```diff
@@ -0,0 +1,206 @@
+    interviewer = "John Smith",
+    interviewStatus = InterviewStatus.NEXTROUND
+)
+
```
To focus on implementing composables, I created this instance of `Interview`

Audienix replied:
Is it temporary and will be removed after the DB implementation will be done?

YANLING-H replied:
Yes, I will remove it once the DB implementation is done.

YANLING-H committed:
Update Color.kt

YANLING-H committed:
fix elevation

YANLING-H committed:
finish

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt:
```diff
@@ -1,48 +1,122 @@
+                        fontWeight = FontWeight.Bold
+                    )
+                }
+                items(25) { PastInterviewCard() }
```
To fix the scrolling, I wrap each horizontal scrollable list inside a item(){} of a `LazyColumn` composable and list the vertical scrollable `PastInterviewCard` directly inside the `LazyColumn` instead of wrap them in a card, but one problem of this is that I can't specify the background color of the the list of pastInterviewCards.

Audienix replied:
That's okay. We will revisit this implementation in future. We are not going to put any separate color for each 3 layout. I have updated the figma file. But card elevation issue is fixed and check my updated implementation.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt:
```diff
@@ -1,48 +1,122 @@
+                    Card(
+                        shape = Shapes.small,
+                        colors = CardDefaults.cardColors(containerColor = DashBoardBackgroundGreen)
+                    ) {
```
To specify the background color of each horizontally scrollable list, I wrapped the lists into `Card` composable.

Audienix replied:
We don't need background color of each horizontally scrollable list, check updated figma and hence we don't need to take card composable. Updated the code.

YANLING-H commented on app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt:
```diff
@@ -1,48 +1,122 @@
-                    navController.navigate(AppScreens.AddInterview.route) {
-                        popUpTo(AppScreens.Dashboard.route)
+            .background(MaterialTheme.colorScheme.background),
+        topBar = { AppBar(stringResource(id = R.string.nav_item_dashboard)) {} },
```
I found a use case where the top bar is given as an attribute, so I removed the Column structure given in the starter code to make the code a bit cleaner,

Commit:
1. Fixed Card Elevation Issue
2. Used string and dimensions from the xmls
3. Added constant class
4. Upgraded compose version
5. Refactored color resources

Commit:
1. Updated DateConverter
